Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -943,6 +939,8 @@ function Get-AzureSqlDatabaseServerResourceId {
         $ResourceDetails = Invoke-RestMethod -Uri $uri -Method $method -Headers $headers -ContentType $script:jsonContentType
         foreach ($resourceDetail in $ResourceDetails.Value) {
             if ($resourceDetail.name -eq $serverName -and $resourceDetail.type -eq $serverType) {
+                $serverNameLower = $serverName.ToLower() 
+                $resourcedetail.id = $resourcedetail.id.Replace($serverName, $serverNameLower)
                 return $resourceDetail.id
```

Comment:
Can you explain the fix in details? Why this change fixes the problem

Reply:
I get the general idea, I'm talking about previous implementation: what was wrong with it?

Reply:
ResourceDetail.Id consists of the servername which should be converted to lowercase, the same we implemented here.
But in the previous change we converted the servername, which wont get reflected to the servername in resourcedetail.id.

Reply:
LGTM